app/controllers/course/courses_controller.rb

```diff
@@ -47,8 +47,7 @@ def load_todos
     return unless current_course_user&.student?
 
     todos = Course::LessonPlan::Todo.pending_for(current_course_user).
-            includes(:user, item: :course).order(updated_at: :desc)
-    # TODO: Fix n+1 query for #can_user_start?
+            preload(:user, { item: [:course, actable: :conditions] }).order(updated_at: :desc)
     todos = todos.select(&:can_user_start?)
```

Comment:
can_user_start? here calls the line below
https://github.com/Coursemology/coursemology2/blob/17bd932080e814a58ecfa50108b355d81d71e76c/app/models/concerns/course/lesson_plan/item_todo_concern.rb#L13

There are 3 types of actable for each actable here (which is why we preload actable):
1. Survey - allow_response_after_end and end_at are part of the survey model
https://github.com/Coursemology/coursemology2/blob/29e08a85b7f410f85fc6cce11722a1e2170802de/app/models/course/survey.rb#L34
2. Video - by default it is always true since can_user_start is not implemented in the video actable class
3. Assessment
https://github.com/Coursemology/coursemology2/blob/268b63fc98ccf65dd95aa5e3d292bfa79e21d8f3/app/models/concerns/course/assessment/todo_concern.rb#L7
The conditions_satisfied_by? below also calls conditions which is why we preload conditions after actable to prevent N+1
https://github.com/Coursemology/coursemology2/blob/e7880d7a08fd1b89ab3133b123bcc91d25866679/lib/extensions/conditional/active_record/base.rb#L60